internal/controller/kustomization_wait_test.go

```diff
@@ -466,3 +558,345 @@ func TestKustomizationReconciler_RESTMapper(t *testing.T) {
 		g.Expect(err).To(HaveOccurred())
 	})
 }
+
+func TestKustomizationReconciler_CancelHealthCheckOnNewRevision(t *testing.T) {
+	g := NewWithT(t)
+	id := "cancel-" + randStringRunes(5)
+	resultK := &kustomizev1.Kustomization{}
+	timeout := 60 * time.Second
+
+	// Enable the CancelHealthCheckOnNewRevision feature
+	originalValue := features.FeatureGates()[features.CancelHealthCheckOnNewRevision]
+	features.FeatureGates()[features.CancelHealthCheckOnNewRevision] = true
+
+	// Initialize the feature gate system properly
+	featGates := feathelper.FeatureGates{}
+	err := featGates.SupportedFeatures(features.FeatureGates())
+	g.Expect(err).NotTo(HaveOccurred(), "failed to initialize feature gates")
+
+	defer func() {
+		features.FeatureGates()[features.CancelHealthCheckOnNewRevision] = originalValue
+	}()
+
+	err = createNamespace(id)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to create test namespace")
+
+	err = createKubeConfigSecret(id)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to create kubeconfig secret")
+
+	// Create initial successful manifests
+	successManifests := []testserver.File{
+		{
+			Name: "configmap.yaml",
+			Body: `apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-config
+  namespace: ` + id + `
+data:
+  foo: bar`,
+		},
+	}
+	artifact, err := testServer.ArtifactFromFiles(successManifests)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	repositoryName := types.NamespacedName{
+		Name:      fmt.Sprintf("cancel-%s", randStringRunes(5)),
+		Namespace: id,
+	}
+
+	err = applyGitRepository(repositoryName, artifact, "main/"+artifact)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	kustomization := &kustomizev1.Kustomization{}
+	kustomization.Name = id
+	kustomization.Namespace = id
+	kustomization.Spec = kustomizev1.KustomizationSpec{
+		Interval: metav1.Duration{Duration: 10 * time.Minute},
+		Path:     "./",
+		Wait:     true,
+		Timeout:  &metav1.Duration{Duration: 30 * time.Second},
```

Comment:
```suggestion
		Timeout:  &metav1.Duration{Duration: 5 * time.Minute},
```
Let's set here a timeout way higher, to make sure the feature gate actually works.